lib/mongo/crypt/auto_encrypter.rb

```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  module Crypt
+
+    # A module that encapsulates auto-encryption functionality
+    #
+    # @api private
+    module AutoEncrypter
+      include Encrypter
+
+      # A Hash of default values for the :extra_options option
+      DEFAULT_EXTRA_OPTIONS = {
+        mongocryptd_uri: 'mongodb://localhost:27020',
+        mongocryptd_bypass_spawn: false,
+        mongocryptd_spawn_path: '',
```

Comment:
Why is this an empty string and not nil?

Reply:
That's a good question -- normally, you'd spawn mongocryptd from your system path, which is why I made the default value of this option an empty string, but I think `mongocryptd` would be a better default value.

Reply:
(My original idea was that you'd probably spawn mongocryptd at a path like `#{mongocryptd_spawn_path}/mongocryptd`, but that's not necessarily a good assumption -- it would be better to let the user specify the entire path.